File: resources/icons/generate_icons.py
#!/usr/bin/env python3
"""
Generate placeholder icons for FrankenBrowser
Creates simple colored placeholder icons for Windows (.ico) and Linux/macOS (.png)
"""

from PIL import Image, ImageDraw, ImageFont
import os

def create_browser_icon(size=256):
    """Create a simple browser icon"""
    # Create a new image with a gradient blue background
    img = Image.new('RGBA', (size, size), (255, 255, 255, 0))
    draw = ImageDraw.Draw(img)

    # Draw a rounded rectangle background (browser window)
    margin = size // 10
    draw.rounded_rectangle(
        [(margin, margin), (size - margin, size - margin)],
        radius=size // 20,
        fill=(66, 135, 245, 255),  # Blue color
        outline=(40, 90, 200, 255),
        width=size // 50
    )

    # Draw address bar
    bar_margin = margin * 2
    bar_height = size // 8
    draw.rounded_rectangle(
        [(bar_margin, bar_margin), (size - bar_margin, bar_margin + bar_height)],
        radius=size // 40,
        fill=(255, 255, 255, 255),
        outline=(200, 200, 200, 255),
        width=2
    )

    # Draw content area
    content_y = bar_margin + bar_height + margin
    draw.rectangle(
        [(bar_margin, content_y), (size - bar_margin, size - margin)],
        fill=(240, 245, 255, 255)
    )

    # Draw some lines to simulate content
    line_margin = bar_margin + size // 20
    line_y = content_y + size // 20
    line_spacing = size // 15
    for i in range(3):
        y = line_y + i * line_spacing
        draw.rectangle(
            [(line_margin, y), (size - bar_margin - size // 20, y + size // 60)],
            fill=(150, 180, 220, 255)
        )

    return img

def save_png(img, path):
    """Save image as PNG"""
    img.save(path, 'PNG')
    print(f"Created PNG: {path}")

def save_ico(img, path):
    """Save image as ICO (Windows icon)"""
    # ICO format supports multiple sizes
    sizes = [(256, 256), (128, 128), (64, 64), (48, 48), (32, 32), (16, 16)]
    images = [img.resize(size, Image.Resampling.LANCZOS) for size in sizes]
    images[0].save(path, format='ICO', sizes=sizes)
    print(f"Created ICO: {path}")

def main():
    script_dir = os.path.dirname(os.path.abspath(__file__))

    # Create the icon
    icon = create_browser_icon(256)

    # Save as PNG for Linux/macOS
    png_path = os.path.join(script_dir, 'browser.png')
    save_png(icon, png_path)

    # Save as ICO for Windows
    ico_path = os.path.join(script_dir, 'browser.ico')
    save_ico(icon, ico_path)

    print("Icon generation complete!")

if __name__ == '__main__':
    main()

File: scripts/generate_icons.rs
#!/usr/bin/env rust-script
//! Generate placeholder icons for FrankenBrowser
//!
//! ```cargo
//! [dependencies]
//! image = "0.24"
//! ```

use image::{ImageBuffer, Rgba, RgbaImage};
use std::path::Path;

fn create_browser_icon(size: u32) -> RgbaImage {
    let mut img = ImageBuffer::new(size, size);

    // Background color - blue
    let bg_color = Rgba([66, 135, 245, 255]);
    let white = Rgba([255, 255, 255, 255]);
    let light_blue = Rgba([240, 245, 255, 255]);
    let line_color = Rgba([150, 180, 220, 255]);

    let margin = size / 10;

    // Fill background
    for y in 0..size {
        for x in 0..size {
            if x >= margin && x < size - margin && y >= margin && y < size - margin {
                img.put_pixel(x, y, bg_color);
            }
        }
    }

    // Draw address bar
    let bar_margin = margin * 2;
    let bar_height = size / 8;
    for y in bar_margin..bar_margin + bar_height {
        for x in bar_margin..size - bar_margin {
            img.put_pixel(x, y, white);
        }
    }

    // Draw content area
    let content_y = bar_margin + bar_height + margin;
    for y in content_y..size - margin {
        for x in bar_margin..size - bar_margin {
            img.put_pixel(x, y, light_blue);
        }
    }

    // Draw content lines
    let line_margin = bar_margin + size / 20;
    let mut line_y = content_y + size / 20;
    let line_spacing = size / 15;
    let line_height = size / 60;

    for _ in 0..3 {
        for y in line_y..line_y + line_height {
            for x in line_margin..size - bar_margin - size / 20 {
                img.put_pixel(x, y, line_color);
            }
        }
        line_y += line_spacing;
    }

    img
}

fn main() {
    let icon_256 = create_browser_icon(256);

    // Save as PNG
    icon_256
        .save("resources/icons/browser.png")
        .expect("Failed to save PNG");
    println!("Created: resources/icons/browser.png");

    // For ICO, save multiple sizes
    // Note: image crate doesn't directly support ICO format,
    // but we'll create a PNG that can be used on all platforms
    println!("Icon generation complete!");
}
